.circleci/config.yml

```diff
@@ -0,0 +1,198 @@
+  gcp-cli: circleci/gcp-cli@2.1.0
+
+commands:
+
+  prepare-tests:
+    description: "prepares tests to run"
+    steps:
+      - run: sed -i .bck s/api_key/$API_KEY/ example/integration_test/app_test.dart
+
+  build-flutter-project:
+    description: "Install dependencies and build"
+    parameters:
+      project:
+        default: .
+        type: string
+    steps:  
+      - run:
+          name: Install Flutter Dependencies
+          command: |
+            cd <<parameters.project>>
+            flutter pub get
+      - run:
+          name: Build
+          command: |
+            cd <<parameters.project>>
+            flutter packages pub run build_runner build
+
+  run-flutter-integration-tests:
+    description: "Run Flutter integration tests"
+    steps:
+      - run:
+          name: Run tests
+          command: |
+            cd example
+            flutter drive --target=integration_test/app_test.dart \
+            --driver=test_driver/integration_test.dart
+
+  gradle-dependencies:
+    description: "Get Gradle dependencies"
+    steps:
+      - run:
+          working_directory: ./android
+          name: Chmod permissions
+          command: sudo chmod +x ./gradlew
+      - restore_cache:
+          name: Restore Gradle cache
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+      - run:
+          working_directory: ./android
+          name: Download Gradle dependencies
+          command: ./gradlew dependencies
+      - save_cache:
+          name: Save Gradle cache
+          paths:
+            - ~/.gradle
+          key: v1-gradle-{{ checksum "./android/build.gradle" }}-{{ checksum  "./android/app/build.gradle" }}-{{ arch }}
+
+  setup-flutter:
+    description: "setup Flutter"
+    steps:
+      - run:
+          name: Download flutter SDK
+          command: git clone -b stable https://github.com/flutter/flutter.git flutter
+      - run:
+          name: Setup environment variables
+          command: echo 'export PATH="$PATH:`pwd`/flutter/bin"'  >> $BASH_ENV
+      - run:
+          name: Flutter doctor
+          command: flutter doctor
+
+jobs:
+  lint:
+    description: "Run static analysis for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Analyze code
+          command: flutter analyze lib
+
+  test:
+    description: "Run tests for Flutter"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - build-flutter-project
+      - run:
+          name: Run tests
+          command: flutter test
+
+  android-integration-test-build:
+    description: "Build Android Flutter integration tests"
+    docker:
+      - image: cirrusci/flutter:stable
+    steps:
+      - checkout
+      - run:
+          name: Change api key
+          command: sed -i s/api_key/$API_KEY/ example/integration_test/app_test.dart
```

Comment:
this should reuse the command defined at the beginning of the file

Reply:
Ok, so I just remembered why I had this different. This runs in a linux machine and it gives issues when I run `-i .bck` with the `.bck` extension. It looks like the syntax is different in macos and linux. I will see if I can come up with something compatible for both